guevaraTorres/assignments/slidingwindow/lc340/lc340.js

```diff
@@ -4,28 +4,106 @@
 
 // You can assume that K is less than or equal to the length of the given string.
 
+/**
+ * 
+ *The window will need to extend while checking to see if there are more than k distinct characters.
+ * When that occurs the window will move up once the earliest character type is removed.
+ * Determining the longest substring(maxLength) while the window slides forward.
+ * 
+ * variables: windowStart, maxLength, characterFrequency, windowEnd
+ * 
+ * edge cases: null, k=0 or ""
+ */
+
 function longest_substring_with_k_distinct(str, k) {
   // TODO: Write code here
-}
-
-console.log(
-  `Length of the longest substring: ${longest_substring_with_k_distinct(
-    'araaci',
-    2
-  )}`
-);
-console.log(
-  `Length of the longest substring: ${longest_substring_with_k_distinct(
-    'araaci',
-    1
-  )}`
-);
-console.log(
-  `Length of the longest substring: ${longest_substring_with_k_distinct(
-    'cbbebi',
-    3
-  )}`
-);
+
+    let windowStart = 0,
+    maxLength = 0,
+    characterFrequency = {};
+
+    if(str.length == 0 || k < 1) return 0;
+
+  //extends the window 
```

Comment:
Instead of in-line comments, write psuedocode above the function declaration. That way the interviewer can read the psuedocode and your code looks neat.